lib/util/http/host-rules.ts

```diff
@@ -120,24 +120,35 @@ export function applyHostRules<GotOptions extends HostRulesGotOptions>(
   const options: GotOptions = { ...inOptions };
   const foundRules = findMatchingRules(options, url);
   const { username, password, token, enabled, authType } = foundRules;
+  let host: string | undefined;
+  try {
+    host = new URL(url).host;
```

Comment:
why not using parseUrl?

https://github.com/renovatebot/renovate/blob/13960240da08a4c85faa5ec898a8b7fb0b9ad648/lib/util/url.ts#L103